x/ethbridge/keeper/metadata.go

```diff
@@ -33,6 +47,48 @@ func (k Keeper) AddTokenMetadata(ctx sdk.Context, metadata types.TokenMetadata)
 	return denomHash
 }
 
+func (k Keeper) AddIBCTokenMetadata(ctx sdk.Context, metadata types.TokenMetadata, cosmosSender sdk.AccAddress) string {
+	logger := k.Logger(ctx)
+	if !IsIBCToken(metadata.Name) {
+		logger.Error("Token is not IBC cannot use this keeper")
+		return ""
+	}
+
+	if !k.oracleKeeper.IsAdminAccount(ctx, cosmosSender) {
+		logger.Error("cosmos sender is not admin account.")
+		return ""
+	}
+
+	denom := k.AddTokenMetadata(ctx, metadata)
```

Comment:
`AddTokenMetadata` should be an internal method to this keeper, so that only functions within this file can call it.

Reply:
This method needs to be called outside this file however. For example the metadata_test unit tests will be unable to run if we do not export this method. Let me know if you still want me to make the changes.

Reply:
This function only needs to be called by the metadata_test externally to this file. Because of this, we should make the function internal so that there are no mix ups where people call this when they shouldn't be able to.